pkg/storage/ingest/pusher.go

```diff
@@ -45,6 +49,14 @@ func newPusherConsumer(p Pusher, reg prometheus.Registerer, l log.Logger) *pushe
 			MaxAge:     time.Minute,
 			AgeBuckets: 10,
 		}),
+		clientErrRequests: promauto.With(reg).NewCounter(prometheus.CounterOpts{
+			Name: "cortex_ingest_storage_reader_client_error_requests_total",
```

Comment:
There's a bit of naming mismatch with the rest of the code. In the other metrics we're calling it "records" and not "requests", which I think makes sense because it allows us to distinguish them with actual requests issued (e.g. requests to Kafka).

WDYT if we have:
- cortex_ingest_storage_reader_records_failed_total: generalised in "failed" and then we differentiate whether it's a client or server error through a label, so we also keep the count of server errors?
- cortex_ingest_storage_reader_records_total

Reply:
done in https://github.com/grafana/mimir/pull/6940/commits/bf4b27d6e32d29cf72136cc6e3af725cb19bf391